search/src/Filtering.cpp

```diff
@@ -46,5 +47,207 @@ std::vector<int> sigmaGFilteredIndices(const std::vector<float>& values,
     }
     return result;
 }
+
+/* Given a set of psi and phi values,
+   return a likelihood value */
+double calculateLikelihoodFromPsiPhi(std::vector<double> psiValues, std::vector<double> phiValues)
+{
+    double psiSum = 0.0;
+    double phiSum = 0.0;
+    
+    for(int i = 0; i < psiValues.size(); i++) {
+        psiSum += psiValues[i];
+        phiSum += phiValues[i];
+        
+    }
+    
+    if(psiSum == 0.0 || phiSum <= 0.0) {
+        return 0.0;
+    }
+    
+    return psiSum / sqrt(phiSum);
+}
+    
+/* Given a set of fluxValues,
+   inverse Phi values, and a
+   vector of positive flux indicies,
+   apply the kalman filters to 
+   said positive values. */
+std::tuple<std::vector<double>, std::vector<double>> calculateKalmanFlux(std::vector<double> fluxValues, 
+                                                                         std::vector<double> vars,
+                                                                         std::vector<int> fluxIdx, bool reverse)
+{
+    int fluxSize = fluxValues.size();
+    double kalmanFluxes[fluxSize];
+    
+    std::vector<double> xhat(fluxSize, 0.0);
+    std::vector<double> p(fluxSize, 0.0);
+    double xhatMinus[fluxSize];
```

Comment:
I'm behind on C++, but why are some variables defined as `std::vector<double>` vs. `double x[size]`?

Reply:
in c++ traditional arrays are a fixed size while std::vectors are dynamically allocated!

Reply:
here it would be more about memory allocation I think. What do you gain by using a c style array?

Reply:
(resolving since I just made these into single double variables anyway lol)